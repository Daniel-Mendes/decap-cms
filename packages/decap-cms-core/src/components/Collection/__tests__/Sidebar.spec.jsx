import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import { vi, describe, it, expect } from 'vitest';
import { fromJS } from 'immutable';

import { Sidebar } from '../Sidebar';

vi.mock('decap-cms-ui-default', async () => {
  const actual = await vi.importActual('decap-cms-ui-default');
  return {
    ...actual,
    Icon: 'mocked-icon',
  };
});

vi.mock('../NestedCollection', () => ({
  default: 'nested-collection',
}));
vi.mock('../CollectionSearch', () => ({
  default: 'collection-search',
}));
vi.mock('../../../actions/collections');

describe('Sidebar', () => {
  const props = {
    searchTerm: '',
    isSearchEnabled: true,
    t: vi.fn(key => key),
  };
  it('should render sidebar with a simple collection', () => {
    const collections = fromJS([{ name: 'posts', label: 'Posts' }]).toOrderedMap();
    const { asFragment, getByTestId } = render(
      <MemoryRouter>
        <Sidebar {...props} collections={collections} />
      </MemoryRouter>,
    );

    expect(getByTestId('posts')).toHaveTextContent('Posts');
    expect(getByTestId('posts')).toHaveAttribute('href', '/collections/posts');

    expect(asFragment()).toMatchSnapshot();
  });

  it('should not render a hidden collection', () => {
    const collections = fromJS([{ name: 'posts', label: 'Posts', hide: true }]).toOrderedMap();
    const { queryByTestId } = render(
      <MemoryRouter>
        <Sidebar {...props} collections={collections} />
      </MemoryRouter>,
    );

    expect(queryByTestId('posts')).toBeNull();
  });

  it('should render sidebar with a nested collection', () => {
    const collections = fromJS([
      { name: 'posts', label: 'Posts', nested: { depth: 10 } },
    ]).toOrderedMap();
    const { asFragment } = render(
      <MemoryRouter>
        <Sidebar {...props} collections={collections} />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render nested collection with filterTerm', () => {
    const collections = fromJS([
      { name: 'posts', label: 'Posts', nested: { depth: 10 } },
    ]).toOrderedMap();
    const { asFragment } = render(
      <MemoryRouter>
        <Sidebar {...props} collections={collections} filterTerm="dir1/dir2" />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render sidebar without search', () => {
    const collections = fromJS([{ name: 'posts', label: 'Posts' }]).toOrderedMap();
    const { asFragment } = render(
      <MemoryRouter>
        <Sidebar {...props} collections={collections} isSearchEnabled={false} />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
