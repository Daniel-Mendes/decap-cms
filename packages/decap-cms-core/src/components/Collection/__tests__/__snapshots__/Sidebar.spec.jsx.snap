// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sidebar > should render nested collection with filterTerm 1`] = `
<DocumentFragment>
  <aside
    class="makeStyles-SidebarContainer-card exupcl73"
  >
    <h2
      class="makeStyles-SidebarHeading exupcl72"
    >
      collection.sidebar.collections
    </h2>
    <collection-search
      collections="OrderedMap { 0: Map { "name": "posts", "label": "Posts", "nested": Map { "depth": 10 } } }"
      searchterm=""
    />
    <ul
      class="makeStyles-SidebarNavList exupcl71"
    >
      <li>
        <nested-collection
          collection="Map { "name": "posts", "label": "Posts", "nested": Map { "depth": 10 } }"
          data-testid="posts"
          filterterm="dir1/dir2"
        />
      </li>
    </ul>
  </aside>
</DocumentFragment>
`;

exports[`Sidebar > should render sidebar with a nested collection 1`] = `
<DocumentFragment>
  <aside
    class="makeStyles-SidebarContainer-card exupcl73"
  >
    <h2
      class="makeStyles-SidebarHeading exupcl72"
    >
      collection.sidebar.collections
    </h2>
    <collection-search
      collections="OrderedMap { 0: Map { "name": "posts", "label": "Posts", "nested": Map { "depth": 10 } } }"
      searchterm=""
    />
    <ul
      class="makeStyles-SidebarNavList exupcl71"
    >
      <li>
        <nested-collection
          collection="Map { "name": "posts", "label": "Posts", "nested": Map { "depth": 10 } }"
          data-testid="posts"
        />
      </li>
    </ul>
  </aside>
</DocumentFragment>
`;

exports[`Sidebar > should render sidebar with a simple collection 1`] = `
<DocumentFragment>
  <aside
    class="makeStyles-SidebarContainer-card exupcl73"
  >
    <h2
      class="makeStyles-SidebarHeading exupcl72"
    >
      collection.sidebar.collections
    </h2>
    <collection-search
      collections="OrderedMap { 0: Map { "name": "posts", "label": "Posts" } }"
      searchterm=""
    />
    <ul
      class="makeStyles-SidebarNavList exupcl71"
    >
      <li>
        <a
          class="makeStyles-SidebarNavLink-sidebarNavLinkActive-SidebarNavLink exupcl70"
          data-testid="posts"
          href="/collections/posts"
        >
          <mocked-icon
            type="write"
          />
          Posts
        </a>
      </li>
    </ul>
  </aside>
</DocumentFragment>
`;

exports[`Sidebar > should render sidebar without search 1`] = `
<DocumentFragment>
  <aside
    class="makeStyles-SidebarContainer-card exupcl73"
  >
    <h2
      class="makeStyles-SidebarHeading exupcl72"
    >
      collection.sidebar.collections
    </h2>
    <ul
      class="makeStyles-SidebarNavList exupcl71"
    >
      <li>
        <a
          class="makeStyles-SidebarNavLink-sidebarNavLinkActive-SidebarNavLink exupcl70"
          data-testid="posts"
          href="/collections/posts"
        >
          <mocked-icon
            type="write"
          />
          Posts
        </a>
      </li>
    </ul>
  </aside>
</DocumentFragment>
`;
